frontend/components/GameVisibilityList.tsx

```diff
@@ -0,0 +1,46 @@
+import * as React from 'react';
+import { connect } from 'react-redux';
+// eslint-disable-next-line no-unused-vars
+import {AppStore} from '../storage/common';
+interface GameVisibilityListProps{
+	gameVisibilityToggleMap: Map<number, GameVisibilityToggleProps>;
+}
+const GamesVisibilityList: React.FC<GameVisibilityListProps> = (props: GameVisibilityListProps) =>{
+	const {gameVisibilityToggleMap: gamesVisibility} = props;
+
+	return (
+		<>
+			{Array.from(gamesVisibility.values()).map(game=> {
+				return <GameVisibilityToggle key={game.id} name={game.name} isVisible={game.isVisible} id={game.id} />;
+			})}
+		</>
+	);
+};
+interface GameVisibilityToggleProps {
+	id: number;
+	name: string;
+	isVisible: boolean;
+}
+
+const GameVisibilityToggle: React.FC<GameVisibilityToggleProps> = (props) => <>
+	<div>
+		<input type="checkbox" checked = {props.isVisible} name={props.name} id={`spoiler #${props.id}`}/> {props.name}
+	</div>
+
+</>;
+
+const mapStateToProps = (state: AppStore): GameVisibilityListProps => {
+	const games = state.games;
```

Comment:
I believe it'd be slightly more readable if you just destructured those two properties with `const {games, gameVisibility} = state;`